classes/robot/crawler.php

```diff
@@ -263,17 +277,21 @@ public function mark_for_crawl($baseurl, $url) {
      *
      * @return timestamp of crawl start
      */
-    public function get_crawlstart() {
-        return property_exists($this->config, 'crawlstart') ? $this->config->crawlstart : 0;
+    public static function get_crawlstart() {
+        self::init_config();
+        $crawlstart = self::$config->crawlstart;
```

Comment:
Yeah, it does not smell good to me too. 
But I just under impression that we stopped at:

1. create static var $config
2. set $config by new static method get_config()
3. remove initialisation from constructor
4. put normalisation code in constructor for settings "crawlstart", "retentionperiod" and "crawlend"

where 1. was for the purpose of retaining the existing $config variable which kept the plugin config and was used in many places in the class.

I actually vote for implementing a static getter methods without utilising the old $config var. I'll do it now.